.github/workflows/maven.yml

```
@@ -17,14 +17,88 @@
       - 'main'
+    types: [opened, synchronize, reopened, labeled]
 
 jobs:
   Build-repo-check:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)